Frontend/UserTech/src/app/interceptors/http-error.interceptor.spec.ts

```diff
@@ -6,106 +6,128 @@ import {NotificationService} from '../services/notification.service';
     checkFunctionCallOnStatus(redirect.redirectToHome, 404);
   });
 
-  it('should redirect to home view on 406 error', ()=>{
+  it('should redirect to home view on 406 error', () => {
     checkFunctionCallOnStatus(redirect.redirectToHome, 406);
   });
 
-  it('should redirect to home view on 403 error', ()=>{
+  it('should redirect to home view on 403 error', () => {
     checkFunctionCallOnStatus(redirect.redirectToHome, 403);
   });
 
-  it('should redirect to home view on 422 error', ()=>{
+  it('should redirect to home view on 422 error', () => {
     checkFunctionCallOnStatus(redirect.redirectToHome, 422);
   });
 
-  it('should send notification with error message on 422 error', ()=>{
+  it('should send notification with error message on 422 error', () => {
     checkFunctionCallOnStatus(notification.error, 422);
     expect(notification.error).toHaveBeenCalledOnceWith('message');
   });
 
-  it('should send notification with error message on 403 error', ()=>{
+  it('should send notification with error message on 403 error', () => {
     checkFunctionCallOnStatus(notification.error, 403);
     expect(notification.error).toHaveBeenCalledOnceWith('message');
   });
 
-  it('should send notification with error message on 406 error', ()=>{
+  it('should send notification with error message on 406 error', () => {
     checkFunctionCallOnStatus(notification.error, 406);
     expect(notification.error).toHaveBeenCalledOnceWith('message');
   });
 
-  it('should send notification on unknown error', ()=>{
+  it('should send notification on unknown error', () => {
     checkFunctionCallOnStatus(notification.error, -1);
   });
 
-  it('should not redirect or send notification if response has #IGNORE_ERROR_INTERCEPT header', ()=>{
+  it('should not redirect or send notification if response has #IGNORE_ERROR_INTERCEPT header', () => {
     request = request.clone({
-        headers: request.headers.set(IGNORE_ERROR_INTERCEPT, '')
+      headers: request.headers.set(IGNORE_ERROR_INTERCEPT, '')
     });
     const next: HttpHandler = {
-        handle(request: HttpRequest<unknown>){
-            return throwError(new HttpErrorResponse({
-                status: 404
-            }));
-    }}
-    interceptor
-        .intercept(request, next)
-        .subscribe(()=>{},()=>{});
+      handle(_: HttpRequest<unknown>): any {
+        return throwError(new HttpErrorResponse({
+          status: 404
+        }));
+      }
+    };
+    interceptor.intercept(request, next).subscribe(() => {}, () => {});
     expect(notification.error).toHaveBeenCalledTimes(0);
     expect(redirect.redirectToHome).toHaveBeenCalledTimes(0);
     expect(redirect.redirectToLogin).toHaveBeenCalledTimes(0);
   });
+
+  it('should not redirect on 422 if path matches constant', () => {
+    (Object.getOwnPropertyDescriptor(router, 'url')?.get as Spy<() => string>).and.returnValue('/rental/return/3');
+    expect(redirect.redirectToHome).toHaveBeenCalledTimes(0);
```

Comment:
Po co to sprawdzamy?

Reply:
Żeby przetestować czy na pewno nie redirectuje jak jesteśmy na odpowiedniej ścieżce, np. /rental/return/1

Reply:
Ok